http://bugzilla.gnome.org/show_bug.cgi?id=350976

https://bugzilla.novell.com/show_bug.cgi?id=146947

libsmbclient returns EINVAL if one sends it filenames with illegal
characters, like '\'.  Display a more useful error message than
"generic error" in that case.

Index: src/file-manager/fm-error-reporting.c
===================================================================
--- src/file-manager/fm-error-reporting.c.orig
+++ src/file-manager/fm-error-reporting.c
@@ -124,6 +124,11 @@ fm_report_error_renaming_file (NautilusF
 							   new_name_truncated);
 			}
 			break;
+	case GNOME_VFS_ERROR_BAD_PARAMETERS:
+		message = g_strdup_printf (_("Couldn't rename \"%s\" to \"%s\".  Please make sure the new name has "
+					     "only valid characters in it."),
+					   original_name_truncated, new_name_truncated);
+		break;
 		default:
 			break;
 		}
